zzz_generated/chip-tool/zap-generated/cluster/Commands.h

```diff
@@ -6874,7 +6874,7 @@ class DeviceEnergyManagementPowerAdjustRequest : public ClusterCommand
     DeviceEnergyManagementPowerAdjustRequest(CredentialIssuerCommands * credsIssuerConfig) :
         ClusterCommand("power-adjust-request", credsIssuerConfig)
     {
-        AddArgument("Power", INT64_MIN, INT64_MAX, &mRequest.power);
+        AddArgument("Power", 0, UINT64_MAX, &mRequest.power);
```

Comment:
Why has ZAP mad this unsigned? Is this a bug in ZAP?

This is the XML which defined Power as "power_mw":

```
    <command source="client" code="0x0000" name="PowerAdjustRequest" optional="true" apiMaturity="provisional">
      <arg name="Power" type="power_mw"/>
      <arg name="Duration" type="elapsed_s"/>
      <description>Allows a client to request an adjustment in the power consumption of an ESA for a specified duration.</description>
    </command>
```

Reply:
Seems like a ZAP bug, yes.  The template has: `{{as_type_min_value type language='c++'}}, {{as_type_max_value type language='c++'}}` but it's clearly ending up with the wrong things...

Reply:
@jamesharrow You want `signed="true"` on your type definitions, no?

Reply:
Ah - yes thanks that has fixed it.